### PR TITLE
show deploy_url in kubernetes resources since it is more useful than …

### PR DESCRIPTION
**Pull request by eatwithforks**
…owner info/repo

/cc @zendesk/samson @zendesk/compute

### CODE REVIEW

File: lib/samson/extra_dig.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def dig_set(keys, value)
       raise ArgumentError, "No key given" if keys.empty?
       keys = keys.dup
       last = keys.pop
-      failed = ->(*) { raise KeyError, "key not found: #{(keys << last).inspect}" }
+      failed = ->(*) { raise KeyError, "key not found: #{keys.inspect}" }
```
Comment (adammw) [inline lib/samson/extra_dig.rb]: seems odd the error message for dig_fetch didn't change...

**Comment by grosser** [comment]
dig_set was wrong, since when setting a,b,c it should say "a,b" did not
exist and not a,b,c since c is never the problem since it is what is being
set

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 2]
```diff
@@ -85,6 +84,15 @@ def build_selector_for_container(container)
       end
     end
 
+    def set_deploy_url
+      templates = [template]
+      templates << pod_template if Kubernetes::RoleConfigFile::PRIMARY_KINDS.include?(template[:kind])
```
Comment (adammw) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]: what about deploy kinds?

File: plugins/kubernetes/test/models/kubernetes/template_filler_test.rb [hunk 3]
```diff
@@ -825,9 +797,9 @@ def secret_annotations(hash)
     end
 
     describe "when something is required" do
-      let(:annotations) { raw_template[:spec][:template][:metadata][:annotations] = {} }
+      let(:pod_annotations) { raw_template[:spec][:template][:metadata][:annotations] = {} }
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/template_filler_test.rb]: probably didn't need to rename these, but it doesn't hurt i guess

**Comment by grosser** [comment]
included in primary

File: plugins/kubernetes/test/models/kubernetes/template_filler_test.rb [hunk 4]
```diff
@@ -161,49 +168,14 @@ def add_init_container(container)
...
       it "only modifies namespace since there is no template" do
         raw_template[:kind] = "ConfigMap"
         raw_template.delete(:spec)
         raw_template[:metadata][:namespace] = 'old'
-        old = raw_template.deep_dup
-        old[:metadata][:namespace] = 'pod1'
-        template.to_hash.must_equal old
+        result = template.to_hash
+        result[:metadata][:namespace].must_equal 'pod1'
+        refute result[:spec]
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/template_filler_test.rb]: i don't really understand this change, is it related?

**Comment by grosser** [comment]
just for consistency to code

**Comment by grosser** [comment]
previously we made no changes to ConfigMap, this now changes since we
always set this annotation

**Pull request merged**